benchmarking/benchmarking.md

```diff
@@ -0,0 +1,103 @@
+# Benchmarking Harness
+
+Author: @biswapanda
+
+# Problem statement:
+In current state, benchmarking has a few problems:
+
+1. UX: Experimentation, debugging, and iteration is hard. 
+use case: As a user, I want to easily experiment with different configs, and get results quickly and compare them.
+
+2. Reproducibility is hard: we don't store the input configs and results.
+use case: As a user, I want to be able to reproduce my experiments and share them with others.
+
+3. benchmarking steps are tightly coupled. If a sinlge step/benchmark config fails the entire process is aborted/retried.   
+
+4. port-forwarding and benchmarking has non-deterministic latency characteristics. 
+
+## Proposed plan:
+
+1. decouple all steps and then compose them together: prep a model, deploy k8s cr, benchmark, collect data 
+
+2. capture configs for the experiment: deploy (config or a reference to deployment), benchmark, model etc
+
+3. we'd run benchmarks inside k8s cluster in k8s native approach.
```

Comment:
Do we still run into k8s DNS issues when trying to solve the "non-deterministic" problem? I think we have this same issue in slurm where we are not guaranteed the nodes are in the same topology. What are your tolerances for "reproducibility?"